### PR TITLE
Eng 7708/fpjs latency

### PR DESCRIPTION
**Pull request by cometman**

### CODE REVIEW

**Review by quacke-nid — DISMISSED** [review]

**Comment by kishisaka-nid** [comment]
Possible to get the wifi/cell connection type in a ct element like the javascript SDK is doing? Android is filling this with either "cell" or "wifi" strings depending on the connection type. I don't think android can get the RTT(round trip time) or effective type. If not no problem I think. They really require the latency. 

  l: latency,
  ct: getConnectionType(),
  rtt: getRTT(),
  et: getEffectiveType(),

**Comment by cometman** [comment]
@kishisaka-nid added connectionType to NIDEvent string

**Review by kishisaka-nid — APPROVED** [review]

**Review by kishisaka-nid — APPROVED** [review]

File: NeuroID/NeuroIDClass/Extensions/NIDLog.swift [hunk 1]
```diff
@@ -178,7 +178,7 @@ func NIDPrintEvent(_ mutableEvent: NIDEvent) {
         case NIDEventName.log.rawValue:
             contextString = "m=\(mutableEvent.m ?? "")"
         case NIDEventName.advancedDevice.rawValue:
-            contextString = "rid=\(mutableEvent.rid ?? "") c=\(mutableEvent.c ?? false)"
+        contextString = "rid=\(mutableEvent.rid ?? "") c=\(mutableEvent.c ?? false) l=\(mutableEvent.l)"
```
Comment (quacke-nid) [inline NeuroID/NeuroIDClass/Extensions/NIDLog.swift:181]: Indentation

File: NeuroID/NeuroIDClass/Extensions/NIDAdvancedDevice.swift [hunk 2]
```diff
@@ -9,14 +9,16 @@ import Foundation
...
+
     static func start(_ advancedDeviceSignals: Bool) -> Bool {
         let started = NeuroID.start()
         
         if !started {
             return started
         }
-       
-        if advancedDeviceSignals {
+               if advancedDeviceSignals {
```
Comment (quacke-nid) [inline NeuroID/NeuroIDClass/Extensions/NIDAdvancedDevice.swift:21]: indentation

**Review by quacke-nid — COMMENTED** [review]
LGTM! Just a few minor nits!

File: NeuroID/NeuroIDClass/Extensions/NIDAdvancedDevice.swift [hunk 3]
```diff
@@ -80,10 +83,12 @@ public extension NeuroID {
         }
     }
     
-    internal static func captureADVEvent(_ requestID: String, cached: Bool) {
+    internal static func captureADVEvent(_ requestID: String, cached: Bool, latency: Double) {
         let nidEvent = NIDEvent(type: .advancedDevice)
         nidEvent.rid = requestID
         nidEvent.c = cached
+        nidEvent.l = latency
+        nidEvent.ct = NetworkMonitoringService.connectionTypeAccessor.rawValue
```
Comment (kevinsites) [inline NeuroID/NeuroIDClass/Extensions/NIDAdvancedDevice.swift:91]: Note:
We have a variable on the NeuroID class `networkMonitor` that is an instance of the `NetworkMonitoringService` class, is there a reason we wouldn't use that with the `connectionType` attribute instead?

File: NeuroID/Utilities/NetworkMonitoringService.swift [hunk 4]
```diff
@@ -18,6 +18,8 @@ internal class NetworkMonitoringService {
 
     internal private(set) var isConnected: Bool = false
     internal private(set) var connectionType: ConnectionType = .unknown
+    
+    internal static var connectionTypeAccessor: ConnectionType = .unknown
```
Comment (kevinsites) [inline NeuroID/Utilities/NetworkMonitoringService.swift:22]: Similar to the above comment, why are we storing the same property twice when we create a class instance?

File: SDKTest/NeuroIDClassTests.swift [hunk 5]
```diff
@@ -34,7 +35,22 @@ class NeuroIDClassTests: XCTestCase {
...
+        let mockService = MockDeviceSignalService()
+        NeuroID.deviceSignalService = mockService
+        _ = NeuroID.configure(clientKey: "key_test_0OMmplsawAp2CQfWrytWA3wL")
+        UserDefaults.standard.removeObject(forKey: Constants.storageAdvancedDeviceKey.rawValue)
+        let randomTimeInMilliseconds = Double(Int.random(in: 0..<3000))
+        mockService.mockResult = .success(("empty mock result. Can be filled with anything", randomTimeInMilliseconds))
+        _ = NeuroID.start(true)
+        let allEvents = DataStore.getAllEvents()
+        let validEvent = allEvents.filter { $0.type == "ADVANCED_DEVICE_REQUEST" }
+        XCTAssertTrue(validEvent.count == 1)
```
Comment (kevinsites) [inline SDKTest/NeuroIDClassTests.swift:51]: NIT: Can we add spacing to separate the setup vs. test vs. assertion for readability?

**Review by kevinsites — CHANGES_REQUESTED** [review]

**Review by kevinsites — APPROVED** [review]

**Review by priya-xavier — APPROVED** [review]

**Pull request merged**